ci.nix

```diff
@@ -0,0 +1,111 @@
+# SPDX-FileCopyrightText: 2020 TBD
+#
+# SPDX-License-Identifier: LicenseRef-Proprietary
+rec {
+  sources = import ./nix/sources.nix;
+  pkgs = import ./nix/nixpkgs-with-haskell-nix.nix;
+  xrefcheck = import sources.xrefcheck;
+  weeder-hacks = import sources.haskell-nix-weeder { inherit pkgs; };
+
+  haskell-nix = pkgs.haskell-nix;
+
+  # all local packages and their subdirectories
+  # we need to know subdirectories to make weeder stuff work
+  local-packages = [{
+    name = "globacap";
+    subdirectory = ".";
+  }];
+
+  # names of all local packages
+  local-packages-names = map (p: p.name) local-packages;
+
+  # haskell.nix package set
+  # parameters:
+  # - release -- 'true' for production build, 'false' for development build
+  # - commitSha, commitDate -- git revision info used during morley-ledgers compilation
+  hs-pkgs = { release, commitSha ? null, commitDate ? null }:
+    haskell-nix.stackProject {
+      src = haskell-nix.haskellLib.cleanGit { src = ./.; };
+
+      modules = [
+        # common options for all local packages:
+        {
+          packages = pkgs.lib.genAttrs local-packages-names (packageName: {
+            package.ghcOptions = with pkgs.lib;
+              concatStringsSep " " ([
+                "-O0"
+                "-Werror"
+              ]
+              # produce *.dump-hi files, required for weeder:
+                ++ optionals (!release) [ "-ddump-to-file" "-ddump-hi" ]);
+            dontStrip = !release; # strip in release mode, reduces closure size
+            doHaddock = !release; # don't haddock in release mode
+
+            # in non-release mode collect all *.dump-hi files (required for weeder)
+            postInstall =
+              if release then null else weeder-hacks.collect-dump-hi-files;
+          });
+        }
+
+        # provide commit sha and date for globacap in release mode:
+        {
+          packages.globacap = {
+            preBuild = ''
+              export MORLEY_DOC_GIT_COMMIT_SHA=${
+                if release then
+                  pkgs.lib.escapeShellArg commitSha
+                else
+                  "UNSPECIFIED"
+              }
+              export MORLEY_DOC_GIT_COMMIT_DATE=${
+                if release then
+                  pkgs.lib.escapeShellArg commitDate
+                else
+                  "UNSPECIFIED"
+              }
+            '';
+          };
+        }
```

Comment:
I would also disable haddock for dependencies, like here: https://gitlab.com/morley-framework/morley/-/blob/e5b7ea86273352e110656b27077e18c3802d5db5/ci.nix#L69. It will make rebuilding dependencies faster